dali/base/dafdesc.cpp

```diff
@@ -1447,7 +1527,13 @@ class CFileDescriptor:  public CFileDescriptorBase, implements ISuperFileDescrip
             attr.setown(createPTreeFromIPT(at));
         else
             attr.setown(createPTree("Attr"));
-
+        if (attr->hasProp("@lfnHash")) // potentially missing for meta coming from a legacy Dali
+            lfnHash = attr->getPropInt("@lfnHash");
+        else if (tracename.length())
+        {
+            lfnHash = hashc((const unsigned char *)tracename.str(), tracename.length(), 0);
```

Comment:
minor: More encapsulated if the function to calculate the hash was in a separately named function - so the actual hash function is isolated.